src/Log/Engine/BaseLog.php

```diff
@@ -94,6 +116,18 @@ public function scopes()
      * @return string
      */
     protected function _format(string $message, array $context = []): string
```

Comment:
I would remove this as well, but it seems like a bigger BC break than removing `_getFormattedDate()`. Ideally, our log engines should only be calling `resolve()` before passing the message to the formatter.

Reply:
I can add the deprecated annotation and tell people to use `resolve()` directly and just suppress the psalm warning.

Reply:
Instead of deprecating this, should this method call resolve and then use the formatter to format? Do engines need to do logic between these two steps?

Reply:
That's what it is doing.

Reply:
I meant that the call to `$this->formatter->format()` would also be in the `_format` method. But looking at it again that isn't great as the level would need to be passed requiring `_format` to change.